storage/range_test.go

```diff
@@ -134,7 +134,13 @@ func (tc *testContext) Start(t *testing.T) {
 	tc.stopper.AddCloser(tc.transport)
 
 	if tc.store == nil {
-		tc.store = NewStore(tc.clock, tc.engine, nil, tc.gossip, tc.transport, TestStoreConfig)
+		ctx := TestStoreContext
+		ctx.Clock = tc.clock
+		ctx.Gossip = tc.gossip
+		ctx.Transport = tc.transport
+		// Dummy DB.
+		ctx.DB = &client.KV{}
```

Comment:
My reasoning behind this change is that doing it this way forces you to consciously pass nonsense for some of the parts that you know you don't care about. If a test did try to use it, it would explode. If we removed the actual check (we likely would have to do the same thing for Gossip, which gets the same treatment), it could be much more subtle to figure out why things are passed as nil, especially because now with a context `nil` is just what you get when you forget something.

If you feel strongly about it, I'll change it.